pkg/resources/secret_with_basic_authentication.go

```diff
@@ -0,0 +1,193 @@
+	client := meta.(*provider.Context).Client
+
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	if err := handleSecretImport(d); err != nil {
+		return nil, err
+	}
+	secretDescription, err := client.Secrets.Describe(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := d.Set("username", secretDescription.Username); err != nil {
+		return nil, err
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func CreateContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	databaseName, schemaName, name := handleSecretCreate(d)
+	id := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
+
+	usernameString := d.Get("username").(string)
+	passwordString := d.Get("password").(string)
+
+	request := sdk.NewCreateWithBasicAuthenticationSecretRequest(id, usernameString, passwordString)
+	if v, ok := d.GetOk("comment"); ok {
+		request.WithComment(v.(string))
+	}
+
+	err := client.Secrets.CreateWithBasicAuthentication(ctx, request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeResourceIdentifier(id))
+
+	return ReadContextSecretWithBasicAuthentication(ctx, d, meta)
+}
+
+func ReadContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	secret, err := client.Secrets.ShowByID(ctx, id)
+	if err != nil {
+		if errors.Is(err, sdk.ErrObjectNotFound) {
+			d.SetId("")
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Warning,
+					Summary:  "Failed to retrieve secret with basic authentication. Target object not found. Marking the resource as removed.",
+					Detail:   fmt.Sprintf("Secret with basic authentication name: %s, Err: %s", id.FullyQualifiedName(), err),
+				},
+			}
+		}
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to retrieve secret.",
+				Detail:   fmt.Sprintf("Secret with basic authentication name: %s, Err: %s", id.FullyQualifiedName(), err),
+			},
+		}
+	}
+	secretDescription, err := client.Secrets.Describe(ctx, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if err = handleSecretRead(d, id, secret, secretDescription); err != nil {
+		return diag.FromErr(err)
+	}
+	if err = d.Set("username", secretDescription.Username); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func UpdateContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	commonSet, commonUnset := handleSecretUpdate(d)
+	set := &sdk.SecretSetRequest{
```

Comment:
are SecretSetRequest and SecretUnsetRequest the same for all the types? If so, I would rather see them passed to handleCommonSecretUpdate (by reference) and setting them inside, WDYT?

Reply:
This is a good option, I've tried to achieve the same result as is done in `api_authentication_integration` but in this case it seems more suitable since comment is the only field to be changed in common